sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/memory.scala

```diff
@@ -201,7 +207,15 @@ case class MemoryStream[A : Encoder](
 
   override def initialOffset: OffsetV2 = LongOffset(-1)
 
+  override def prepareForTriggerAvailableNow(): Unit = synchronized {
+    availableNowEndOffset = latestOffset(initialOffset, ReadLimit.allAvailable())
+  }
+
   override def latestOffset(): OffsetV2 = {
+    throw new IllegalStateException("Should not reach here!")
+  }
```

Comment:
Not sure why this cannot be called. Cannot `MemoryStream` be used as a `MicroBatchStream` after this change?

Reply:
When data source implements AdmissionControl, this method is guaranteed to be never called. latestOffset(start, limit) will supersede this.